webapp/blog/logic.py

```diff
@@ -16,7 +16,32 @@ def strip_excerpt(raw_html):
     return html.unescape(cleantext).replace("\n", "")
 
 
-def transform_article(article, featured_image=None, author=None):
+def replace_images_with_cloudinary(content):
+    """Prefixes images with cloudinary optimised URLs and adds srcset for
+    image scaling
+
+    :param content: The HTML string to convert
+
+    :returns: Update HTML string with converted images
+    """
+    cloudinary = (
+        "https://res.cloudinary.com/" "canonical/image/fetch/q_auto,f_auto,"
+    )
+    return re.sub(
```

Comment:
This is really hard to read and understand just by looking at it like this...
Is there no other way to make it simplier

Reply:
To try to make this more readable, how about something like:

```suggestion
    cloudinary = 'https://res.cloudinary.com/'

    urls=[
        cloudinary + 'canonical/image/fetch/q_auto,f_auto,w_1120/\g<url>',
        cloudinary + 'canonical/image/fetch/q_auto,f_auto,w_750/\g<url>',
        cloudinary + 'canonical/image/fetch/q_auto,f_auto,w_960/\g<url>',
        cloudinary + 'canonical/image/fetch/q_auto,f_auto,w_1120/\g<url>',
    ]
    
    image_match = r'<img(?P<prefix>[^>]*) src="(?P< url>[^"]+)"(?P<suffix>[^>]*)>'
    replacement = (
        '<img\g<prefix>'
        ' src="{urls[0]}"'
        ' srcset="{urls[1]} 750w, {urls[2]} 960w, {urls[3]} 1120w"'
        ' sizes="(max-width: 375px) 560px, (max-width: 480px) 880px, 1120px"'
        '\g<suffix>>'
    ).format(urls)

    return re.sub(image_match, replacement, content)
```

(I've not tested this code)

Reply:
I'll try your suggestion out locally nottrobin, if it works I'll accept it :+1: